custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/util/MavenHelper.java

```diff
@@ -64,25 +67,50 @@ public int run(File buildDir, boolean failOnError, String ... args) throws IOExc
     }
 
     public boolean artifactExistsInLocalCache(DependencyInfo dep, String version, String packaging) {
-        String path = String.format("~/.m2/repository/%s/%s/%s/%s-%s.%s",
-                dep.groupId.replaceAll("\\.", "/"),
-                dep.artifactId,
-                version,
-                dep.artifactId,
-                version,
-                packaging);
-        LOGGER.log(Level.INFO, "Checking {0}", path);
+        final String folder = "repository";
+        String path = getDependencyPath(folder, dep, version, packaging);
         File expectedFile = new File(path);
+        LOGGER.log(Level.INFO, "Checking {0}", expectedFile.getAbsolutePath());
         return expectedFile.exists();
     }
 
+    private static String getDependencyPath(final String folder, final DependencyInfo dep, final String version, final String packaging) {
+        return String.format("%s/.m2/%s/%s/%s/%s/%s-%s.%s",
```

Comment:
Maven repository can be actually located in another place, but it's a separate story